entity-framework/core/what-is-new/ef-core-9.0/breaking-changes.md

```diff
@@ -20,33 +20,170 @@ EF Core 9 targets .NET 8. This means that existing applications that target .NET
+| [The `id` property no longer contqins the discriminator by default](#cosmos-id-property-changes)                                                     | High       |
+| [Sync I/O via the Azure Cosmos DB provider is no longer supported](#cosmos-nosync)                                                                   | Medium     |
+| [SQL queries must now project JSON values directly](#cosmos-sql-queries-with-value)                                                                  | Medium     |
+| [Undefined results are now automatically filtered from query results](#cosmos-undefined-filtering)                                                   | Medium     |
+| [Incorrectly translated queries are no longer translated](#cosmos-incorrect-translations)                                                            | Medium     |
+| [`HasIndex` now throws instead of being ignored](#cosmos-hasindex-throws)                                                                            | Low        |
+| [`IncludeRootDiscriminatorInJsonId` was renamed to `HasRootDiscriminatorInJsonId` after 9.0.0-rc.2](#cosmos-IncludeRootDiscriminatorInJsonId-rename) | Low        |
+
+### High-impact changes
+
+<a name="cosmos-discriminator-name-change"></a>
+
+#### The discriminator property is now named `$type` instead of `Discriminator`
+
+[Tracking Issue #34269](https://github.com/dotnet/efcore/issues/34269)
+
+##### Old behavior
+
+EF automatically adds a discriminator property to JSON documents to identify the entity type that the document represents. In previous versions of EF, this JSON property used to be named `Discriminator` by default.
+
+##### New behavior
+
+Starting with EF Core 9.0, the discriminator property is now called `$type` by default. If you have existing documents in Cosmos DB from previous versions of EF, these use the old `Discriminator` naming, and after upgrading to EF 9.0, queries against those documents will fail.
+
+##### Why
+
+An emerging JSON practice uses a `$type` property in scenarios where a document's type needs to be identified. For example, .NET's System.Text.Json also supports polymorphism, using `$type` as its default discriminator property name ([docs](/dotnet/standard/serialization/system-text-json/polymorphism#customize-the-type-discriminator-name)). To align with the rest of the ecosystem and make it easier to interoperate with external tools, the default was changed.
+
+##### Mitigations
+
+The easiest mitigation is to simply configure the name of the discriminator property to be `Discriminator`, just as before:
+
+```csharp
+modelBuilder.Entity<Session>().HasDiscriminator<string>("Discriminator");
+```
+
+Doing this for all your top-level entity types will make EF behave just like before.
+
+At this point, if you wish, you can also update all your documents to use the new `$type` naming.
+
+TODO: Cosmos SDK code sample for updating the documents
+
```

Comment:
👀

Reply:
To really help people out with these two high-impact breaking changes, we should ideally include some code with the Cosmos SDK (not EF) to just rename `Discriminator` -> `$type`, and remove the discriminator from `id`... @AndriySvyryd is this something you're with doing, or maybe can see if someone else in the team can?

Reply:
Filed https://github.com/dotnet/EntityFramework.Docs/issues/4821